pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -215,9 +219,14 @@ public static class ControllerPeriodicTasksConf {
         "controller.realtime.segment.deepStoreUploadRetryEnabled";
     public static final String DEEP_STORE_RETRY_UPLOAD_TIMEOUT_MS =
         "controller.realtime.segment.deepStoreUploadRetry.timeoutMs";
+    public static final String ENABLE_TMP_SEGMENT_ASYNC_DELETION =
+        "controller.realtime.split.commit.segment.tmp.cleanup.async.enable";
```

Comment:
Based on this config we should either run or skip the `RealtimeTempSegmentCleaner`. I think that's not wired right now.

Reply:
good suggestion

Reply:
deleting tmp segments will be invoked with the existing realtime segment validation manager